src/services/EdFi.DataManagementService.Api/ApiSchema/Extensions/JsonHelperExtensions.cs

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using Json.Path;
+
+namespace EdFi.DataManagementService.Api.ApiSchema.Extensions;
+
+public static class JsonHelperExtensions
+{
+    /// <summary>
+    /// Helper to go from a scalar JSONPath selection directly to the selected JsonNode,
+    /// or null if the node does not exist
+    /// </summary>
+    public static JsonNode? SelectNodeFromPath(this JsonNode jsonNode, string jsonPathString, ILogger logger)
+    {
+        try
+        {
+            JsonPath? jsonPath = JsonPath.Parse(jsonPathString);
+            if (jsonPath == null)
+            {
+                logger.LogError("Malformed JSONPath string '{jsonPathString}'", jsonPathString);
+                throw new InvalidOperationException($"Malformed JSONPath string '{jsonPathString}'");
```

Comment:
No need to log the error here. Middleware will take care of the logging.

Should this be critical / deadly?